src/System.Management.Automation/engine/regex.cs

```diff
@@ -142,7 +142,11 @@ StringComparison GetStringComparison()
                 {
                     stringComparison = Options.HasFlag(WildcardOptions.CultureInvariant)
                         ? StringComparison.InvariantCultureIgnoreCase
-                        : StringComparison.CurrentCultureIgnoreCase;
+                        : CultureInfo.CurrentCulture.Name.Equals("en-US-POSIX", StringComparison.OrdinalIgnoreCase)
```

Comment:
Should we add a test case?

We have one for.. hashtable.

Reply:
https://github.com/PowerShell/PowerShell/blob/1ce2de4a15836a2b57d84b115d74039e53d6b056/test/powershell/Language/Parser/LanguageAndParser.TestFollowup.Tests.ps1#L307

Reply:
Test added. Two things to be noted:
1. Setting the current culture to be 'en-us-POSIX' doesn't do anything on Windows -- `[string]::Equals("a", "A", [System.StringComparison]::CurrentCultureIgnoreCase)` always returns `true`.
2. It seems to be the existing way of changing culture works fine. The following script writes out `false` as expected.
```
try {
      $oldCulture = [System.Globalization.CultureInfo]::CurrentCulture
      [System.Globalization.CultureInfo]::CurrentCulture = [System.Globalization.CultureInfo]::new('en-US-POSIX')
      [string]::Equals("a", "A", [System.StringComparison]::CurrentCultureIgnoreCase)
}
finally {
     [System.Globalization.CultureInfo]::CurrentCulture = $oldCulture
}
```